xarray/core/computation.py

```diff
@@ -597,7 +597,7 @@ def broadcast_compat_data(
         data = duck_array_ops.transpose(data, order)
 
     if new_dims != reordered_dims:
-        key_parts = []
+        key_parts: list[Union[slice, None]] = []
```

Comment:
FYI this list is python3.9+, probably fine but maybe safer to use `List`?

```suggestion
        key_parts: List[Optional[slice]] = []
```

Reply:
> FYI this list is python3.9+

Ah right, yes let's use `List`